nixos/modules/programs/appgate-sdp.nix

```diff
@@ -0,0 +1,84 @@
+{ config, pkgs, lib, ... }:
+
+with lib;
+
+{
+  options = {
+    programs.appgate-sdp = {
+      enable = mkEnableOption
+        "AppGate SDP VPN client";
+    };
+  };
+
+  config = mkIf config.programs.appgate-sdp.enable {
+
+    environment.systemPackages = [ pkgs.appgate-sdp ];
+    services.dbus.packages = [ pkgs.appgate-sdp ];
+    boot.kernelModules = [ "tun" ];
+
+    systemd.services = {
+      appgatedriver = {
+        description = "AppGate driver service";
+        wantedBy = [ "multi-user.target" ];
+        serviceConfig = {
+          Restart = "always";
+          Type = "forking";
+          ExecStartPre = ''
```

Comment:
Maybe you want to move this into `preStart`?

Reply:
Indeed it does include unit files, but I'm not sure how to wrap those.